### ISSUE TITLE
importing static fields from classes using @:build doesn't work

### ISSUE THREAD

**Issue by ousado** [HaxeFoundation/haxe#2729]
Test.hx:

```
import Test.OtherClass.*;
// import Test.OtherClass.foo;  // fails directly on import

@:publicFields
@:build(Macro.build())
class OtherClass {
    static var foo = 123;   
}

class Test {
    static function main(){
        trace(foo);            // Unknown identifier : foo
    }
}
```

Macro.hx:

```
import haxe.macro.Context;
import haxe.macro.Expr;

class Macro {
    static function build():Array<Field>{
        var fields = Context.getBuildFields();
        trace(fields);
        return fields;
    }
}
```

I'd expect OtherClass.foo to be in scope in Test.main().

**Comment by Simn** [comment]
Fixed by #11168.

**Comment by ousado** [comment]
wow!
